library/core/benches/iter.rs

```diff
@@ -367,3 +367,28 @@ fn bench_partial_cmp(b: &mut Bencher) {
 fn bench_lt(b: &mut Bencher) {
     b.iter(|| (0..100000).map(black_box).lt((0..100000).map(black_box)))
 }
+
+// Ideally the compiler should be able to optimize away the range checks here and turn those
```

Comment:
Isn't better to add a codegen test?
https://github.com/rust-lang/rust/tree/673d0db5e393e9c64897005b470bfeb6d5aec61b/src/test/codegen

Reply:
It's easier to run benchmarks (on stage 0), but sure I can add one.